ingestion/src/main/java/feast/store/serving/redis/RedisCustomIO.java

```diff
@@ -224,55 +212,57 @@ public WriteDoFn withTimeout(int timeout) {
 
       @Setup
       public void setup() {
-        jedis = new Jedis(host, port, timeout);
+        this.ingestionClient.setup();
       }
 
       @StartBundle
       public void startBundle() {
+        try {
+          ingestionClient.connect();
```

Comment:
Should we perhaps make sure that we do not unnecessarily reconnect? One of the issues we previously had is that too many connections were being established from ingestion pipelines to Redis. In the order of thousands.

Reply:
yeah. in the implementation, I will just check before the connection object before creating another one.